base/linalg/triangular.jl

```diff
@@ -1854,6 +1919,166 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
 end
 logm(A::LowerTriangular) = logm(A.').'
 
+# Auxiliary functions for logm and matrix power
+
+# Compute accurate diagonal of A = A0^s - I
+#   Al-Mohy, "A more accurate Briggs method for the logarithm",
+#      Numer. Algorithms, 59, (2012), 393–402.
+function sqrt_diag!(A0::UpperTriangular, A::UpperTriangular, s)
+    n = checksquare(A0)
+    @inbounds for i = 1:n
+        a = complex(A0[i,i])
+        if s == 0
+            A[i,i] = a - 1
+        else
+            s0 = s
+            if imag(a) >= 0 && real(a) <= 0 && a != 0
+                a = sqrt(a)
+                s0 = s - 1
+            end
+            z0 = a - 1
+            a = sqrt(a)
+            r = 1 + a
+            for j = 1:s0-1
+                a = sqrt(a)
+                r = r * (1 + a)
+            end
+            A[i,i] = z0 / r
+        end
+    end
+end
+
+# Repeatedly compute the square roots of A so that in the end its
+# eigenvalues are close enough to the positive real line
+function invsquaring(A0::UpperTriangular, theta)
+    maxsqrt = 100
+    tmax = size(theta, 1)
+    n = checksquare(A0)
+    A = complex(copy(A0))
+    p = 0
+    m = 0
+
+    # Compute repeated roots
+    d = complex(diag(A))
+    dm1 = similar(d, n)
+    s = 0
+    for i = 1:n
+        dm1[i] = d[i] - 1
```

Comment:
I looked at the paper, and it is also not dimensionally correct.  I have no doubt that it produces the right answer, but it is still unsatisfying to have an algorithm where the number of iterations depends on an overall scaling (and it will cause trouble with arrays of dimensionful quantities).

My instict would be to compute `A^p = α^p (A/α)^p`, where `α = norm(A,1)` is a scale factor that makes  `A/α` dimensionless and scale-invariant.  That is, apply the Higham algorithm to `A/α`.  (The cost of the additional scaling step should be negligible.)